gensim/models/wrappers/wordrank.py

```diff
@@ -0,0 +1,204 @@
+        `wr_path` is the path to the Wordrank directory.
+        `corpus_file` is the filename of the text file to be used for training the Wordrank model.
+        Expects file to contain space-separated tokens in a single line
+        `out_path` is the path to directory which will be created to save embeddings and training data.
+        `size` is the dimensionality of the feature vectors.
+        `window` is the number of context words to the left (and to the right, if symmetric = 1).
+        symmetric` if 0, only use left context words, else use left and right both.
+        `min_count` = ignore all words with total frequency lower than this.
+        `max_vocab_size` upper bound on vocabulary size, i.e. keep the <int> most frequent words. Default is 0 for no limit.
+        `sgd_num` number of SGD taken for each data point.
+        `lrate` is the learning rate (too high diverges, give Nan).
+        `period` is the period of xi variable updates
+        `iter` = number of iterations (epochs) over the corpus.
+        `epsilon` is the power scaling value for weighting function.
+        `dump_period` is the period after which embeddings should be dumped.
+        `reg` is the value of regularization parameter.
+        `alpha` is the alpha parameter of gamma distribution.
+        `beta` is the beta parameter of gamma distribution.
+        `loss` = name of the loss (logistic, hinge).
+        `memory` = soft limit for memory consumption, in GB.
+        `cleanup_files` if to delete directory and files used by this wrapper, setting to False can be useful for debugging
+        `sorted_vocab` = if 1 (default), sort the vocabulary by descending frequency before assigning word indexes.
+        `ensemble` = 0 (default), use ensemble of word and context vectors
+        """
+
+        meta_data_path = 'matrix.meta'
+        vocab_file = 'vocab.txt'
+        temp_vocab_file = 'tempvocab.txt'
+        cooccurrence_file = 'cooccurrence'
+        cooccurrence_shuf_file = 'wiki.toy'
+        meta_file = 'meta'
+        
+        # prepare training data (cooccurrence matrix and vocab)
+        model_dir = os.path.join(wr_path, out_path)
+        meta_dir = os.path.join(model_dir, 'meta')
+        os.makedirs(meta_dir)
+        logger.info("Dumped data will be stored in '%s'", model_dir)
+        copyfile(corpus_file, os.path.join(meta_dir, corpus_file.split('/')[-1]))
+        os.chdir(meta_dir)
+
+        cmd0 = ['../../glove/vocab_count', '-min-count', str(min_count), '-max-vocab', str(max_vocab_size)]
+        cmd1 = ['../../glove/cooccur', '-memory', str(memory), '-vocab-file', temp_vocab_file, '-window-size', str(window), '-symmetric', str(symmetric)]
+        cmd2 = ['../../glove/shuffle', '-memory', str(memory)]
+        cmd3 = ['cut', '-d', " ", '-f', '1', temp_vocab_file]
+        cmds = [cmd0, cmd1, cmd2, cmd3]
+        logger.info("Preparing training data using glove code '%s'", cmds)
+        o0 = smart_open(temp_vocab_file, 'w')
```

Comment:
It's safer to open files in binary mode (`wb`), and explicitly encode all strings written there.

Has this been tested on unicode (non-ASCII) data?